packages/core/src/prompt/index.ts

```diff
@@ -0,0 +1,326 @@
+    if (config.thinking) {
+      /*
+       * Add chain of thought if enabled
+       */
+      if (config.thinking.autoChainOfThought) {
+        component.push("Let's solve this step by step \n");
+      }
+      /*
+       * Add steps if specified
+       */
+      if (config.thinking.steps?.length) {
+        config.thinking.steps.forEach((step, index) => {
+          component.push(`${index + 1}. ${step}`);
+        });
+      }
+
+      /*
+       * Add tree of thought if enabled
+       */
+      if (config.thinking.autoTreeOfThought) {
+        component.push(
+          "\nLet's explore multiple ways of solving this problem and go through each approach step by step",
+        );
+      } else if (config.thinking.branches) {
+        /*
+         * Else add explicit three of thought branches
+         */
+        Object.entries(config.thinking.branches).forEach(([branch, thoughts]) => {
+          component.push(`\n${branch} Approach:`);
+          thoughts.forEach((thought, index) => {
+            component.push(`  ${index + 1}. ${thought}`);
+          });
+        });
+      }
+
+      /*
+       * Add self-ask if enabled
+       */
+      if (config.thinking.autoSelfAsk) {
+        component.push("\nLet's break this down by asking ourselves questions \n");
+      }
+      /*
+       * Add verification if enabled
+       */
+      if (config.thinking.autoVerification) {
+        component.push("\nLet's verify our solution \n");
+      }
+
+      /*
+       * Add verification steps if specified
+       */
+      if (config.thinking.verificationSteps?.length) {
+        component.push("\nLet's verify our solution");
+        config.thinking.verificationSteps.forEach((step, index) => {
+          component.push(`${index + 1}. ${step}`);
+        });
+      }
+    }
+
+    return component;
+  }
+
+  get layout() {
+    const intro: string[] = [];
+    const outro: string[] = [];
+
+    /*
+     * Intro
+     */
+
+    /*
+     * Add persona if specified
+     */
+    if (this.config.persona) {
+      intro.push(`As a ${this.config.persona.toUpperCase()}:`);
+    }
+
+    /*
+     * Add style and tone if specified
+     */
+    if (this.config.style || this.config.tone) {
+      const styleStr = [this.config.style, this.config.tone].filter(Boolean).join(', ');
+      intro.push(`Write in a ${styleStr} manner: \n`);
+    }
+
+    /*
+     * Add perspective if specified
+     */
+    if (this.config.perspective) {
+      intro.push(`From the perspective of ${this.config.perspective}: \n`);
+    }
+
+    intro.push(`${this.config.intent} \n`);
+
+    /*
+     * Main
+     */
+    const main = this.buildPromptComponent(this.config);
+
+    /*
+     * Outro
+     */
+    if (this.config.outputFormat) {
+      outro.push(`Please output your response in the following format: ${this.config.outputFormat} \n`);
+    }
+
+    return {
+      intro,
+      main,
+      outro,
+    };
+  }
+
+  component(config: Partial<PromptTemplateComponent>) {
```

Comment:
I'm not sure if it's better but I would create an API like so
 ```suggestion
  prompt(prompt: string, config: Partial<PromptTemplateComponent>) {
```

Reply:
Where I guess the prompt is the thing you really need. Also do we need input? Can't we use string literals instead in my prompt? or is there value to define these in LLMs?